app/models/account.rb

```diff
@@ -107,6 +107,10 @@ def blocking?(other_account)
     blocking.include?(other_account)
   end
 
+  def domain_blocking?(other_account)
+    AccountDomainBlock.where(account: self, domain: other_account.domain).exists?
```

Comment:
Account could probably use a has_many for :account_domains_blocks, which this method could then use.